compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java

```diff
@@ -2065,6 +2065,260 @@ public void exitStringTemplateLiteral(BallerinaParser.StringTemplateLiteralConte
+    }
+
+    @Override
+    public void exitSelectClause(BallerinaParser.SelectClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        boolean isSelectAll = ctx.MUL() == null ? false : true;
+        boolean isGroupByClauseAvailable = ctx.groupByClause() == null ? false : true;
+        boolean isHavingClauseAvailable = ctx.havingClause() == null ? false : true;
+        this.pkgBuilder.endSelectClauseNode(isSelectAll, isGroupByClauseAvailable, isHavingClauseAvailable,
+                getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void enterSelectExpressionList(BallerinaParser.SelectExpressionListContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startSelectExpressionList();
+    }
+
+    @Override
+    public void exitSelectExpressionList(BallerinaParser.SelectExpressionListContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.endSelectExpressionList(getWS(ctx), ctx.getChildCount() / 2 + 1);
+    }
+
+    @Override
+    public void enterWhereClause(BallerinaParser.WhereClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startWhereClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void exitWhereClause(BallerinaParser.WhereClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.endWhereClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void enterWindowClause(BallerinaParser.WindowClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startWindowClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void exitWindowClause(BallerinaParser.WindowClauseContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.endWindowsClauseNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void enterStreamingInput(BallerinaParser.StreamingInputContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        this.pkgBuilder.startStreamingInputNode(getCurrentPos(ctx), getWS(ctx));
+    }
+
+    @Override
+    public void exitStreamingInput(BallerinaParser.StreamingInputContext ctx) {
+        if (ctx.exception != null) {
+            return;
+        }
+
+        boolean isWindowAvailable = false;
+        boolean isfirstWhereClauseAvailable = false;
+        boolean isSecondWhereClauseAvailable = false;
+        isWindowAvailable = ctx.windowClause() == null ? false : true;
+        isfirstWhereClauseAvailable = ctx.whereClause(0) == null ? false : true;
+        isSecondWhereClauseAvailable = ctx.whereClause(1) == null ? false : true;
```

Comment:
For above 3 statements, no need to have variable declaration in a separate line.  Also simplify the check like below. 
`boolean isWindowAvailable = ctx.windowClause() != null`

Reply:
fixed.